src/charm.py

```diff
@@ -214,6 +217,17 @@ def _on_collect_unit_status(self, event: ops.CollectStatusEvent) -> None:
         if not self.model.relations.get("cos-agent"):
             event.add_status(BlockedStatus("Grafana Agent is not related"))
 
+        upstream_snap = get_installed_snap_service(UPSTREAM_SNAP)
+
+        # this is necessary when doing a charm upgrade coming from revision 27
+        if upstream_snap.present:
+            event.add_status(
+                BlockedStatus(
+                    "Please add an empty file as resource. For more info see: "
+                    "https://charmhub.io/openstack-exporter#known-issues"
+                )
```

Comment:
Juju's messages are usually clipped when it's too long, so I am not sure if the link will be shown in `juju status`

Reply:
Before it wasn't, but then I changed to fit.